cpp/src/community/k_truss_impl.cuh

```diff
@@ -656,265 +764,76 @@ k_truss(raft::handle_t const& handle,
                                                    *vertex_partition_range_lasts);
     }
     renumber_map = std::move(tmp_renumber_map);
+
   }
 
-  // 5. Decompress the resulting graph to an edges list and ind intersection of edges endpoints
-  // for each partition using detail::nbr_intersection
+  // 4. Compute triangle count using nbr_intersection and unroll weak edges
 
   {
     auto cur_graph_view = modified_graph_view ? *modified_graph_view : graph_view;
-    rmm::device_uvector<vertex_t> edgelist_srcs(0, handle.get_stream());
-    rmm::device_uvector<vertex_t> edgelist_dsts(0, handle.get_stream());
-    std::optional<rmm::device_uvector<edge_t>> num_triangles{std::nullopt};
-    std::optional<rmm::device_uvector<weight_t>> edgelist_wgts{std::nullopt};
 
     edge_weight_view =
       edge_weight ? std::make_optional((*edge_weight).view())
                   : std::optional<edge_property_view_t<edge_t, weight_t const*>>{std::nullopt};
 
-    auto prop_num_triangles = edge_triangle_count<vertex_t, edge_t, false>(handle, cur_graph_view);
+    cugraph::edge_property_t<decltype(cur_graph_view), bool> edge_mask(handle, cur_graph_view);
+    cugraph::fill_edge_property(handle, cur_graph_view, edge_mask.mutable_view(), bool{true});
 
-    std::tie(edgelist_srcs, edgelist_dsts, edgelist_wgts, num_triangles, std::ignore) =
-      decompress_to_edgelist(
+    std::chrono::seconds s (0);             // 1 second
```

Comment:
Two thoughts...
 1. Have you looked at https://github.com/rapidsai/cugraph/blob/branch-24.08/cpp/include/cugraph/utilities/high_res_timer.hpp which I think you could just include and use rather than reinventing.  There are features to manage multiple timers inside that as well
 2. Probably should remove these before we merge the PR

Reply:
I only wanted to time the triangle count part and I got a compiler error when copying the `high_res_timer` timing code from our tests so I didn't spend much time understanding what was wrong. But I will look into it again. 

> Probably should remove these before we merge the PR

I only have them for benchmarking on Draco. I will remove them in my next commit today along with all the unused functors